### PR TITLE
#167484541 - Setup e2e tests for the application

### PR DESCRIPTION
**Pull request by malaba6**
#### What does this PR do?
- Sets up end to end tests for the application

#### Description of Task to be done?
- Install cypress
- User cypress integration to test the UI

#### How should this be manually tested?
1. Clone this branch.
2. Do `npm install`
3. Do `npm start` and you should see the landing page.
4. Add this virtual environment `SERVER_URL=http://localhost:3000` to the `.env` file
5. In another terminal window run `npm run e2e:test` to run the end to end test

#### Screenshots
Screen Shot 2019-09-25 at 11 29 32

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,9 +5,10 @@
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test",
     "test:CI": "CI=true react-scripts test \"-- --coverage\"",
-    "eject": "react-scripts eject"
+    "test": "react-scripts test --watchAll",
+    "eject": "react-scripts eject",
+    "e2e:test": "cypress open"
```
Comment (placideirandora) [inline package.json]: Why didn't you use the normal creact-react-app test script?

Comment (harerakalex) [reply package.json]: +1 @placideirandora

Comment (malaba6) [reply package.json]: @placide the normal `react-script test` does not do end-to-end testing, it tests the react components. Just to make clear, end-to-end testing means, testing the UI, which the normal testing does not do

Comment (placideirandora) [reply package.json]: So, you mean that we will be testing normal endpoints (src) and the UI (public)?

Comment (malaba6) [reply package.json]: No we will be testing React components (normal testing) and the actual User Interface (not the `public` folder), For example, let's say that when you click a button, you expect it's text to change. We will use `Cypress` to test if the text within the button actually changed after the button click.

Comment (placideirandora) [reply package.json]: Indeed! You meant two kinds of testing: react and cypress.

**Review by clemwek — APPROVED** [review]
LGTM

**Pull request merged**